CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.9)
+
+project(VulkanMemoryAllocator)
+
+set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
```

Comment:
1. Would this affect projects which do insource builds? While convenient for safeties sake, its good to not enforce something if not necessary. If it doesn't propagate *out* of the VMA CMakeLists.txt file, this isn't an issue.
2. Should this go before or after the `project()` call?
3. These are undocumented features (that I only learned about today). Anything that is intentionally undocumented make me wary of using them, as I have no idea how to determine what they do without either playing with it extensively or asking someone who does know.
4. This doesn't prevent a CMakeCache.txt or CMakeFiles/ folders from being created either.

Reply:
Since I could not find clear documentation about whether this command propagates or not, I removed it.